### PR TITLE
[develop] Lucero - Adds driver structure for writing tests.

### PR DESCRIPTION
**Pull request by luqp**

### CODE REVIEW

File: src/main/java/org/fundacionjala/core/ui/config/Environment.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    private static final String EXPLICIT_TIME_WAIT = "explicitTimeWait";
+    private static final String IMPLICIT_TIME_WAIT = "implicitTimeWait";
+    private static final String REDUCE_EXPLICIT_TIME = "reduceExplicitTime";
+
+    private static Environment instance;
+
+    private final Properties properties;
+    private Map<String, Object> envConfig;
+
+
```
Comment (Snip-77) [inline src/main/java/org/fundacionjala/core/ui/config/Environment.java]: Unnecessary space here

File: gradle.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+platformName=android
+realMobile=true
+
+#time waits
+implicitTimeWait=60
+explicitTimeWait=20
+reduceExplicitTime=2
```
Comment (Snip-77) [inline gradle.properties]: Did you meant "reduce**d**ExplicitTime"?

File: src/main/java/org/fundacionjala/core/ui/device/DeviceBuilder.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.fundacionjala.core.ui.device;
+
+import org.fundacionjala.core.ui.config.Environment;
+
+public final class DeviceBuilder {
+
+    private static Device device = Environment.getInstance().isRealMobile()
+            ? new Real()
```
Comment (Snip-77) [inline src/main/java/org/fundacionjala/core/ui/device/DeviceBuilder.java]: I don't understand this xD

File: src/main/java/org/fundacionjala/core/ui/device/DeviceFactory.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package org.fundacionjala.core.ui.device;
+
+import org.fundacionjala.core.ui.config.Environment;
+
+public final class DeviceFactory {
```
Comment (Snip-77) [inline src/main/java/org/fundacionjala/core/ui/device/DeviceFactory.java:5]: Why do we need a DeviceBuilder and a DeviceFactory? Can't we have only one?

File: src/main/java/org/fundacionjala/core/ui/driver/Android.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import org.openqa.selenium.WebDriver;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.fundacionjala.core.ui.device.DeviceFactory.getDevice;
+
+public final class Android extends AbstractPlatform {
+
+    private static final String IP_ADDRESS = "127.0.0.1";
+    private static final int PORT = 4723;
```
Comment (Snip-77) [inline src/main/java/org/fundacionjala/core/ui/driver/Android.java]: Maybe we could get this from gradle.properties

**Review by Snip-77 — COMMENTED** [review]
Good abstraction! Have some comments tho.

File: src/main/java/org/fundacionjala/core/ui/config/Environment.java (hunk 1 above)
Comment (luqp) [reply src/main/java/org/fundacionjala/core/ui/config/Environment.java]: Done

File: gradle.properties (hunk 2 above)
Comment (luqp) [reply gradle.properties]: Done

File: src/main/java/org/fundacionjala/core/ui/device/DeviceFactory.java (hunk 4 above)
Comment (luqp) [reply src/main/java/org/fundacionjala/core/ui/device/DeviceFactory.java:5]: Done

File: src/main/java/org/fundacionjala/core/ui/device/DeviceBuilder.java (hunk 3 above)
Comment (luqp) [reply src/main/java/org/fundacionjala/core/ui/device/DeviceBuilder.java]: Is a Ternary Operator
`variable = (condition) ? expressionTrue :  expressionFalse;`

File: src/main/java/org/fundacionjala/core/ui/driver/Android.java (hunk 5 above)
Comment (luqp) [reply src/main/java/org/fundacionjala/core/ui/driver/Android.java]: Done

**Review by Snip-77 — APPROVED** [review]

**Pull request merged**